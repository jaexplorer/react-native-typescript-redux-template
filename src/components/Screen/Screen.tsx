import React, { FC } from 'react';
import { View } from 'react-native';
import { useTheme } from '../../lib/Theme';
import stylesCreator from './ScreenStyles';

const Screen: FC<ScreenProps> = ({ children, backgroundColor }) => {
  const [styles] = useTheme(stylesCreator, { backgroundColor });
  return (
    <View testID="root" style={[styles.screenContainer]}>
      <View style={styles.contentContainer}>{children}</View>
    </View>
  );
};

export interface ScreenProps {
  backgroundColor?: string;
}

export default Screen;
