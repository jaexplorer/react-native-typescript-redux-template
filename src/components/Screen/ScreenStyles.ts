import { StyleSheet } from 'react-native';
import { StylesCreator } from '../../lib/Theme';

const stylesCreator: StylesCreator = (theme, scale, moderateScale, props) =>
  StyleSheet.create({
    screenContainer: {
      flex: 1,
      maxWidth: theme.SCREEN_WIDTH,
      maxHeight: theme.SCREEN_HEIGHT,
      justifyContent: 'center',
      alignContent: 'center',
      alignItems: 'center',
      padding: theme.SCREEN_MARGIN,
      backgroundColor: props.backgroundColor || theme.colours.backgroundColours.primary,
    },
    contentContainer: {
      width: theme.CONTENT_WIDTH,
      height: theme.CONTENT_HEIGHT,
      justifyContent: 'center',
      alignContent: 'center',
      alignItems: 'center',
    },
  });

export default stylesCreator;
