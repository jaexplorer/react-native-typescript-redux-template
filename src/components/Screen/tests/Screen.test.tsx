import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';
import Screen from '../Screen';

test('Does component render test', () => {
  const { baseElement } = render(
    <Screen>
      <View />
    </Screen>,
  );
  expect(baseElement).toMatchSnapshot();
});
