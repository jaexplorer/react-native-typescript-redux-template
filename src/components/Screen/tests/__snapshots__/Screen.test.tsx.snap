// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does component render test 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#F5F5F5",
            "flex": 1,
            "justifyContent": "center",
            "maxHeight": 1334,
            "maxWidth": 750,
            "padding": 64.28571428571428,
          },
        ]
      }
      testID="root"
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "height": 1269.7142857142858,
            "justifyContent": "center",
            "width": 685.7142857142858,
          }
        }
      >
        <View />
      </View>
    </View>
  </View>
</View>
`;
